x-pack/plugins/infra/public/alerting/metric_threshold/components/custom_equation/custom_equation_editor.tsx

```diff
@@ -0,0 +1,211 @@
+const VAR_NAMES = range(65, 65 + 26).map((c) => String.fromCharCode(c));
+
+export const CustomEquationEditor: React.FC<CustomEquationEditorProps> = ({
+  onChange,
+  expression,
+  fields,
+  aggregationTypes,
+  errors,
+}) => {
+  const [customMetrics, setCustomMetrics] = useState<CustomMetrics>(
+    expression?.customMetrics ?? [NEW_METRIC]
+  );
+  const [label, setLabel] = useState<string | undefined>(expression?.label || undefined);
+  const [equation, setEquation] = useState<string | undefined>(expression?.equation || undefined);
+  const debouncedOnChange = useMemo(() => debounce(onChange, 500), [onChange]);
+
+  const handleAddNewRow = useCallback(() => {
+    setCustomMetrics((previous) => {
+      const currentVars = previous?.map((m) => m.name) ?? [];
+      const name = first(xor(VAR_NAMES, currentVars)) || 'XX'; // This should never happen.
+      const nextMetrics = [...(previous || []), { ...NEW_METRIC, name }];
+      debouncedOnChange({ ...expression, customMetrics: nextMetrics, equation, label });
+      return nextMetrics;
+    });
+  }, [debouncedOnChange, equation, expression, label]);
+
+  const handleDelete = useCallback(
+    (name: string) => {
+      setCustomMetrics((previous) => {
+        const nextMetrics = previous?.filter((row) => row.name !== name) ?? [NEW_METRIC];
+        const finalMetrics = (nextMetrics.length && nextMetrics) || [NEW_METRIC];
+        debouncedOnChange({ ...expression, customMetrics: finalMetrics, equation, label });
+        return finalMetrics;
+      });
+    },
+    [equation, expression, debouncedOnChange, label]
+  );
+
+  const handleChange = useCallback(
+    (metric: MetricExpressionCustomMetric) => {
+      setCustomMetrics((previous) => {
+        const nextMetrics = previous?.map((m) => (m.name === metric.name ? metric : m));
+        debouncedOnChange({ ...expression, customMetrics: nextMetrics, equation, label });
+        return nextMetrics;
+      });
+    },
+    [equation, expression, debouncedOnChange, label]
+  );
+
+  const handleEquationChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setEquation(e.target.value);
+      debouncedOnChange({ ...expression, customMetrics, equation: e.target.value, label });
+    },
+    [debouncedOnChange, expression, customMetrics, label]
+  );
+
+  const handleLabelChange = useCallback(
+    (e: React.ChangeEvent<HTMLInputElement>) => {
+      setLabel(e.target.value);
+      debouncedOnChange({ ...expression, customMetrics, equation, label: e.target.value });
+    },
+    [debouncedOnChange, expression, customMetrics, equation]
+  );
+
+  const disableAdd = useMemo(() => customMetrics?.length === 26, [customMetrics]);
+
+  const filteredAggregationTypes = omit(aggregationTypes, OMITTED_AGGREGATIONS_FOR_CUSTOM_METRICS);
+
+  const metricRows = customMetrics?.map((row) => {
+    const disableDelete = customMetrics?.length === 1;
```

Comment:
It seems that this `disableDelete` could be assigned outside the `map` function, since it only checks if `customMetrics?.length` is equals to 1. It doesn't use anything from `row`.